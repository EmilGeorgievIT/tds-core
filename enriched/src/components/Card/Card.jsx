import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import { warn } from '../../deprecate';

if (process.env.BROWSER) {
  require('./Card.scss');
}

const Card = ({ className, children, ...rest }) => {
  if (className) {
    warn('Card', 'Custom CSS classes are deprecated. This component does not support custom styling.');
  }

  if (rest.style) {
    warn('Card', 'Inline styles are deprecated. This component does not support custom styling.');
  }

  return (
    <div {...rest} className={classNames('card', className)}>
      {children}
    </div>
  );
};

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Card;
