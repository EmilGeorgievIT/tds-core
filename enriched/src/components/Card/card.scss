@import '~telus-thorium-core/scss/settings/variables';

.card {
  border: 0;
  box-sizing: border-box;
  border-radius: 4px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.08);
  background-color: $color-white;
  padding: 30px;
}
