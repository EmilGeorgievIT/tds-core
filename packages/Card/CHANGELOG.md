# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.8...@tds/core-card@2.1.0) (2019-09-12)


### Features

* **core-card:** add new variants, deprecate current variants ([afc1eab](https://github.com/telusdigital/tds/commit/afc1eab))





## [2.0.8](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.7...@tds/core-card@2.0.8) (2019-09-06)

**Note:** Version bump only for package @tds/core-card





## [2.0.7](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.6...@tds/core-card@2.0.7) (2019-08-22)

**Note:** Version bump only for package @tds/core-card





## [2.0.6](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.5...@tds/core-card@2.0.6) (2019-08-09)


### Bug Fixes

* **core-card:** change box-shadow to black ([2eb6ed1](https://github.com/telusdigital/tds/commit/2eb6ed1))





## [2.0.5](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.4...@tds/core-card@2.0.5) (2019-08-01)

**Note:** Version bump only for package @tds/core-card





## [2.0.4](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.3...@tds/core-card@2.0.4) (2019-07-30)

**Note:** Version bump only for package @tds/core-card





## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.2...@tds/core-card@2.0.3) (2019-07-25)

**Note:** Version bump only for package @tds/core-card





## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.1...@tds/core-card@2.0.2) (2019-07-17)

**Note:** Version bump only for package @tds/core-card





## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-card@2.0.0...@tds/core-card@2.0.1) (2019-06-25)

**Note:** Version bump only for package @tds/core-card





# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.11...@tds/core-card@2.0.0) (2019-06-18)


### Features

* **core-card:** upgrade to styled components ([a7d7b6d](https://github.com/telusdigital/tds/commit/a7d7b6d))


### BREAKING CHANGES

* **core-card:** styled-components now a peer dependency





## [1.0.11](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.10...@tds/core-card@1.0.11) (2019-06-04)

**Note:** Version bump only for package @tds/core-card

## [1.0.10](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.9...@tds/core-card@1.0.10) (2019-05-01)

**Note:** Version bump only for package @tds/core-card

## [1.0.9](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.8...@tds/core-card@1.0.9) (2019-04-23)

**Note:** Version bump only for package @tds/core-card

## [1.0.8](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.7...@tds/core-card@1.0.8) (2019-04-22)

**Note:** Version bump only for package @tds/core-card

## [1.0.7](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.6...@tds/core-card@1.0.7) (2019-04-11)

### Bug Fixes

- **core-card:** use className prop for Box ([a9e713c](https://github.com/telusdigital/tds/commit/a9e713c))

## [1.0.6](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.5...@tds/core-card@1.0.6) (2019-03-18)

**Note:** Version bump only for package @tds/core-card

## [1.0.5](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.4...@tds/core-card@1.0.5) (2019-03-18)

**Note:** Version bump only for package @tds/core-card

## [1.0.4](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.3...@tds/core-card@1.0.4) (2019-03-11)

**Note:** Version bump only for package @tds/core-card

## [1.0.3](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.2...@tds/core-card@1.0.3) (2019-03-05)

**Note:** Version bump only for package @tds/core-card

## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.1...@tds/core-card@1.0.2) (2019-02-06)

**Note:** Version bump only for package @tds/core-card

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-card@1.0.0...@tds/core-card@1.0.1) (2018-04-02)

### Bug Fixes

- **core-box:** update tests ([c0f0343](https://github.com/telusdigital/tds/commit/c0f0343))
