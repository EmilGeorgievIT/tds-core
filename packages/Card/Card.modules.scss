@import '../../src/scss/settings/colours';

.card {
  composes: none rounded from '../../src/components/Borders.modules.scss';
}

.white {
  composes: card;

  box-shadow: 0px 0px 16px 0px rgba(0, 0, 0, 0.1);
  background-color: $color-white;
}

.lavender {
  composes: card;

  background-color: $color-white-lilac;
}

.grey {
  composes: card;

  background-color: $color-athens-grey;
}
