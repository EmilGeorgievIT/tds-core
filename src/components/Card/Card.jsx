import React from 'react'
import PropTypes from 'prop-types'

import Box from '../Box/Box'
import safeRest from '../../utils/safeRest'

import styles from './Card.modules.scss'

/**
 * A container that serves as an entry point to more detailed information.
 */
const Card = ({ variant, children, ...rest }) => (
  <Box
    {...safeRest(rest)}
    spacing="padding"
    horizontal={3}
    vertical={4}
    dangerouslyAddClassName={styles[variant]}
  >
    {children}
  </Box>
)

Card.propTypes = {
  variant: PropTypes.oneOf(['white', 'lavender', 'gray']).isRequired,
  /**
   * The content. Can be text, any HTML element, or any component.
   */
  children: PropTypes.node.isRequired,
}

Card.defaultProps = {
  variant: 'white',
}

export default Card
