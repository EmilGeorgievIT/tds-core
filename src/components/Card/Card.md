Cards will stretch to fill their container. Wrap the card in another element to apply size.

```jsx
<Card>
  <Heading level="h2">Need a hand?</Heading>
  
  <div style={{ margin: "1rem 0" }}>
    <Paragraph>
      Ready to order? Have a question? We'll get back to you, with volume discounts available to larger accounts.
    </Paragraph>
  </div>
  
  <ButtonLink href="#">Request sales callback</ButtonLink>
</Card>
```

### Coloured variants

Card also supports a 'grey' or 'lavender' `variant`. These appear flat, while the default white Card appears floating.

```jsx {"props": {"className": "docs__layout-horizontally"}}
<div className="wrapper">
  <Card variant="lavender">
    <Heading level="h4">Holiday deal</Heading>
    
    <div style={{ margin: "1rem 0" }}>
      <Paragraph size="medium">Get a new smartphone for $0<br/>on a 2-year plan.</Paragraph>
    </div>
  </Card>
  
  <Card variant="grey">
    <Heading level="h4">Find the right gift</Heading>
    
    <div style={{ margin: "1rem 0" }}>
      <Paragraph>Find something they'll love from our<br/>selection of great devices.</Paragraph>
    </div>
    
    <ChevronLink variant="primary" href="">Explore latest devices</ChevronLink>
  </Card>
</div>
```
