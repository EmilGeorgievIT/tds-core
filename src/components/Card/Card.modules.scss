@import '../../scss/settings/colours.scss';

.card {
  composes: none rounded from '../Borders.modules.scss';

  padding: 30px;
}

.white {
  composes: card;

  box-shadow: 0 0 16px 0 rgba(213, 213, 213, 0.5);
  background-color: $color-white;
}

.lavender {
  composes: card;

  background-color: $color-lavender-blush;
}

.gray {
  composes: card;

  background-color: $color-athens-grey;
}
