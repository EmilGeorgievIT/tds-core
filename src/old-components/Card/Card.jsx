import React from 'react'
import PropTypes from 'prop-types'

import classNames from 'classnames'

import { deprecate } from '../../warn'

import './Card.scss'

/**
 * A container that serves as an entry point to more detailed information.
 */
const Card = ({ className, children, ...rest }) => {
  if (className) {
    deprecate('Card', 'Custom CSS classes are deprecated. This component does not support custom styling.')
  }

  if (rest.style) {
    deprecate('Card', 'Inline styles are deprecated. This component does not support custom styling.')
  }

  return (
    <div {...rest} className={classNames('card', className)}>
      {children}
    </div>
  )
}

Card.propTypes = {
  /**
   * One or more CSS class names separated by spaces to append onto the container.
   * @deprecated since v0.18.0. Card will no longer support custom styling.
   */
  className: PropTypes.string,
  /**
   * The Card's content. Can be text, any HTML element, or any component.
   */
  children: PropTypes.node.isRequired
}

export default Card
