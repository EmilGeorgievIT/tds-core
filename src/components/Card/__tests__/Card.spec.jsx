import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Card from '../Card'

describe('<Card />', () => {
  const doShallow = (props = {}) => shallow(<Card {...props}>Some content</Card>)

  it('renders', () => {
    const card = doShallow()

    expect(toJson(card)).toMatchSnapshot()
  })

  it('can be presented as one of the allowed variants', () => {
    let card = doShallow()
    expect(card).toHaveClassName('white')

    card = doShallow({ variant: 'white' })
    expect(card).toHaveClassName('white')

    card = doShallow({ variant: 'lavender' })
    expect(card).toHaveClassName('lavender')

    card = doShallow({ variant: 'gray' })
    expect(card).toHaveClassName('gray')
  })

  it('passes additional attributes to the input element', () => {
    const card = doShallow({ role: 'some-role', 'data-some-value': 'some value' })

    expect(card).toHaveProp('role', 'some-role')
    expect(card).toHaveProp('data-some-value', 'some value')
  })

  it('does not allow custom CSS', () => {
    const card = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(card).not.toHaveProp('className', 'my-custom-class')
    expect(card).not.toHaveProp('style')
  })
})
