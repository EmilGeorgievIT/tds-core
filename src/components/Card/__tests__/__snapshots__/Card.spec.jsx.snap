// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders 1`] = `
<div
  class="white"
  x="3"
  y="4"
>
  Children
</div>
`;
